### PR TITLE
Added map functionality to user tour when marker is clicked.

### PR DESCRIPTION
**Pull request by hstrodtman**
FYI First marker clicked does not close automatically like the others.

### CODE REVIEW

**Comment by KBeas21** [comment]
I saw some Geolocation stuff did you also update the map to use better geolocation? I know Scott was talking about that at some point.

**Comment by hstrodtman** [comment]
> I saw some Geolocation stuff did you also update the map to use better geolocation? I know Scott was talking about that at some point.

No, this map uses the same geolocation/direction service that Scott set up. The changes I made were to add a waypoints array that is populated with the lat/long coordinates of each garden in the tour and then updated the getDirections function to add a stop for each waypoint and set the destination back to the user's location.

**Review by KBeas21 — APPROVED** [review]

File: src/main/resources/templates/gardens/tour.html [hunk 1]
```diff
@@ -53,10 +53,118 @@ <h5 class="card-title" th:text="${garden.name}" ></h5>
...
+        });
+        marker.addListener('click', function() {
+            infoWindow.setContent('<p>' +
+                gardenAddress +
+                '</p>');
+            getDirections(userPosition);
+            infoWindow.open(map, marker);
+        });
+    };
+
```
Comment (MulesKC) [inline src/main/resources/templates/gardens/tour.html:149]: I added the following code here to get the directions to somewhat auto-appear and then that also fixes the first window that didn't close when another opens issue: 
    map.addListener('mouseover', function() {
        getDirections(userPosition);
    });

Let me know if it works for you!

File: src/main/resources/templates/gardens/tour.html [hunk 2]
```diff
@@ -53,10 +53,118 @@ <h5 class="card-title" th:text="${garden.name}" ></h5>
...
+            position: gardenLatLng,
+            map: map,
+            icon: 'https://img.icons8.com/color/48/000000/water-wheel.png',
+            animation: google.maps.Animation.DROP
+        });
+        marker.addListener('click', function() {
+            infoWindow.setContent('<p>' +
+                gardenAddress +
+                '</p>');
+            getDirections(userPosition);
```
Comment (MulesKC) [inline src/main/resources/templates/gardens/tour.html]: would remove this if you like that code idea below

File: src/main/resources/templates/gardens/tour.html (hunk 1 above)
Comment (KBeas21) [reply src/main/resources/templates/gardens/tour.html:149]: Also, watch your spacing here. this function is actually in the create marker function so it should be indented over.

**Review by jamiander — APPROVED** [review]

**Pull request merged**